java-checks/src/main/java/org/sonar/java/checks/IgnoredReturnValueCheck.java

```diff
@@ -90,6 +90,9 @@
   public void visitNode(Tree tree) {
     ExpressionTree expr = ((ExpressionStatementTree) tree).expression();
     if (expr.is(Tree.Kind.METHOD_INVOCATION)) {
+      if (isSwitchExpressionYieldedExpression(tree)) {
```

Comment:
Can we do this check conditionally depending on the current java version? To avoid FN such as:
```java
switch("cond") {
  default:
    s.replace("a", "b");
}
```